src/SprykerEco/Yves/Braintree/Form/DataProvider/CheckoutShipmentFormDataProvider.php

```diff
@@ -0,0 +1,211 @@
+use Generated\Shared\Transfer\ShipmentTransfer;
+use Spryker\Shared\Kernel\Store;
+use Spryker\Shared\Kernel\Transfer\AbstractTransfer;
+use Spryker\Shared\Money\Dependency\Plugin\MoneyPluginInterface;
+use Spryker\Yves\StepEngine\Dependency\Form\StepEngineFormDataProviderInterface;
+use SprykerEco\Yves\Braintree\Dependency\Client\BraintreeToGlossaryClientInterface;
+use SprykerEco\Yves\Braintree\Dependency\Client\BraintreeToShipmentClientInterface;
+use SprykerShop\Yves\CheckoutPage\Form\Steps\ShipmentForm;
+
+class CheckoutShipmentFormDataProvider implements StepEngineFormDataProviderInterface
+{
+    public const FIELD_ID_SHIPMENT_METHOD = 'idShipmentMethod';
+
+    /**
+     * @var \SprykerEco\Yves\Braintree\Dependency\Client\BraintreeToShipmentClientInterface
+     */
+    protected $shipmentClient;
+
+    /**
+     * @var \SprykerEco\Yves\Braintree\Dependency\Client\BraintreeToGlossaryClientInterface
+     */
+    protected $glossaryClient;
+
+    /**
+     * @var \Spryker\Shared\Kernel\Store
+     */
+    protected $store;
+
+    /**
+     * @var \Spryker\Shared\Money\Dependency\Plugin\MoneyPluginInterface
+     */
+    protected $moneyPlugin;
+
+    /**
+     * @param \SprykerEco\Yves\Braintree\Dependency\Client\BraintreeToShipmentClientInterface $shipmentClient
+     * @param \SprykerEco\Yves\Braintree\Dependency\Client\BraintreeToGlossaryClientInterface $glossaryClient
+     * @param \Spryker\Shared\Kernel\Store $store
+     * @param \Spryker\Shared\Money\Dependency\Plugin\MoneyPluginInterface $moneyPlugin
+     */
+    public function __construct(
+        BraintreeToShipmentClientInterface $shipmentClient,
+        BraintreeToGlossaryClientInterface $glossaryClient,
+        Store $store,
+        MoneyPluginInterface $moneyPlugin
+    ) {
+        $this->shipmentClient = $shipmentClient;
+        $this->glossaryClient = $glossaryClient;
+        $this->store = $store;
+        $this->moneyPlugin = $moneyPlugin;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return \Generated\Shared\Transfer\QuoteTransfer
+     */
+    public function getData(AbstractTransfer $quoteTransfer)
+    {
+        if ($quoteTransfer->getShipment() === null) {
+            $shipmentTransfer = new ShipmentTransfer();
+            $quoteTransfer->setShipment($shipmentTransfer);
+        }
+
+        return $quoteTransfer;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return array
+     */
+    public function getOptions(AbstractTransfer $quoteTransfer)
+    {
+        return [
+            ShipmentForm::OPTION_SHIPMENT_METHODS => $this->createAvailableShipmentChoiceList($quoteTransfer),
+        ];
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\QuoteTransfer $quoteTransfer
+     *
+     * @return array
+     */
+    protected function createAvailableShipmentChoiceList(QuoteTransfer $quoteTransfer)
```

Comment:
I'm not leaving comments about typehints for public methods, that can be declared in interfaces, but protected one should have return type for sure